test/integration/Session.test.js

```diff
@@ -0,0 +1,89 @@
+import assert from 'assert'
+
+import StreamrClient from '../../src'
+import config from './config'
+
+describe('Session', () => {
+    let clientApiKey
+    let clientPrivateKey
+    let clientUsernamePassword
+    let clientNone
+
+    const createClient = (opts = {}) => new StreamrClient({
+        url: config.websocketUrl,
+        restUrl: config.restUrl,
+        autoConnect: false,
+        autoDisconnect: false,
+        ...opts,
+    })
+
+    beforeAll(() => {
+        clientApiKey = createClient({
```

Comment:
Missing a test case for incorrect credentials (should lead to `loginFunction` promise rejected)